### ISSUE TITLE
Presence / Private channel events are not forwarded to channel subscribers

### ISSUE THREAD

**Issue by elbojoloco** [soketi/pws#14]
I'm using a basic setup for local development using Laravel as backend. Broadcasting regular public channels works fine, but when I try to use private or presence channels, the events don't seem to be forwarded to the subscribers. When I replace the credentials with a real pusher app everything works fine, this together with public channels functioning verifies that it's not a misconfigured app issue. I also have DEBUG enabled in my pWS server but it's not logging connections or events, which would really help with debugging. 

My Laravel pusher config looks like this (default values are used):
```
        'pusher' => [
            'driver' => 'pusher',
            'key' => env('PUSHER_APP_KEY', 'dev'),
            'secret' => env('PUSHER_APP_SECRET', 'dev'),
            'app_id' => env('PUSHER_APP_ID', '1'),
            'options' => [
                'cluster' => env('PUSHER_APP_CLUSTER', 'eu'),
                'host' => '127.0.0.1',
                'port' => 6001,
                'scheme' => 'http'
            ],
        ],
```

My client code looks like this:

```
        this.#socket = new Echo({
            broadcaster: 'pusher',
            cluster: 'eu',
            key: 'dev',
            wsHost: '127.0.0.1',
            wsPort: 6001,
            forceTLS: false,
            auth: {
                headers: {
                    'X-GPCHAT-UUID': this.#uuid,
                    'X-GPCHAT-NAME': this.#userName,
                }
            }
        });

        this.#socket.channel('test-channel').listen('TestEvent', console.log) // This works

        this.#socket
            .join(`App.Models.Chat.${this.#chatId}`) // This works
            .here(participants => {) // This works
                console.log(participants);
            })
            .joining(participant => {) // This works
                console.log(participant);
            })
            .leaving(participant => {) // This works
                console.log(participant);
            })
            .listen('MessageSent', (e) => { // This is never called
                console.log(e)
            })
```

I also tried different variations of the listeners:
`MessageSent`, `.MessageSent`, `App\\Events\\MessageSent` or `.App\\Events\\MessageSent`.

I also tried using the Pusher-js client library directly to listen for Events, no success:
```
        let client = new Pusher('dev', {
            wsHost: '127.0.0.1',
            wsPort: 6001,
            forceTLS: false,
            disableStats: true,
            enabledTransports: ['ws'],
            authEndpoint: '/broadcasting/auth',

            auth: {
                headers: {
                    'X-GPCHAT-UUID': this.#uuid,
                    'X-GPCHAT-NAME': this.#userName,
                }
            }
        });

        client.subscribe('presence-App.Models.Chat.3').bind('App\\Events\\MessageSent', (message) => { // Doesn't work
            console.log(message)
        });
        client.subscribe('test-channel').bind('App\\Events\\TestEvent', (message) => { // Works
            console.log(message)
        });
```

Keep in mind that in both examples the authentication for the presence channel is successful and transmitted to the pWS server:
![image](https://user-images.githubusercontent.com/7912315/124487830-7365d180-ddaf-11eb-8b29-cfc72535b39e.png)

Any ideas how to correctly implement presence/private channels?

**Comment by elbojoloco** [comment]
After hours of debugging, I finally found some more info on this issue. 

My MessageSent event looks like this:

```
<?php

namespace App\Events;

use App\Models\Message;
use Illuminate\Broadcasting\Channel;
use Illuminate\Broadcasting\InteractsWithSockets;
use Illuminate\Broadcasting\PresenceChannel;
use Illuminate\Contracts\Broadcasting\ShouldBroadcast;
use Illuminate\Foundation\Events\Dispatchable;
use Illuminate\Queue\SerializesModels;

class MessageSent implements ShouldBroadcast
{
    use Dispatchable, InteractsWithSockets, SerializesModels;

    /**
     * @var \App\Models\Message
     */
    public $message;

//    public $body;

    public function __construct(Message $message)
    {
        $this->message = $message;

        $this->message->load('participant:id');

//        $this->body = 'message';
    }

    public function broadcastOn()
    {
        return new PresenceChannel('chats');
    }
}
```

This does not seem to be forwarded to subscribers, but when I make `$message` a private property (laravel only forwards public properties when broadcasting) it DOES end up at the client! This is so weird to me.. It doesn't seem to have to do with the length of the data, because I tried sending a string of 10k characters length and that also went through. There must be something in the pWS that doesn't like the model structure or something?

**Comment by elbojoloco** [comment]
When I hop into the source and disable the authMiddleware all my events are coming through.
```
    events(res) {
        this.attachMiddleware(res, [
            this.jsonBodyMiddleware,
            this.corsMiddleware,
            this.appMiddleware,
            // this.authMiddleware,
            this.broadcastEventRateLimitingMiddleware,
        ]).then(res => {
```

But as explained above, i consistently reproduce a working/not working event by changing the event payload.. I am at a loss for words here. Here is a working event:

```
    public function __construct(Message $message)
    {
        $this->message = $message;

        // $this->message->load('participant:id,name');
    }

    public function broadcastOn()
    {
        return new PresenceChannel('chats.' . $this->message->chat_id);
    }
```

Note the lazy loaded relationship is commented out

This event does NOT work:
```
    public function __construct(Message $message)
    {
        $this->message = $message;

        $this->message->load('participant:id,name');
    }

    public function broadcastOn()
    {
        return new PresenceChannel('chats.' . $this->message->chat_id);
    }
```

**Comment by elbojoloco** [comment]
Okay sorry for the spam, but I have found the issue. If the payload contains a valid URL, then the `authMiddleware` will not accept it. That's it.

**Comment by rennokki** [comment]
@elbojoloco Sorry for the delay. This got fixed in #22, the explanation is in #21. Latest version has the fix.

**Comment by elbojoloco** [comment]
@rennokki You're a legend, this package is such a life saver.
